server/command.js

```diff
@@ -0,0 +1,243 @@
+    root = user;
+  } else if (root === '@' || root === 'here') {
+    root = user.location;
+  } else {
+    throw Error("Can't happen.  Regex is too liberal.");
+  }
+  if (!suffix) {
+    return root;
+  }
+  // Clear no-longer needed local variables so they don't leak into the eval.
+  str = undefined;
+  m = undefined;
+  return eval('(root)' + suffix);
+};
+
+
+$.utils = {};
+$.utils.command = {};
+
+$.utils.command.prepositions = Object.create(null);
+$.utils.command.prepositions['with'] = 'with/using';
+$.utils.command.prepositions['using'] = 'with/using';
+$.utils.command.prepositions['at'] = 'at/to';
+$.utils.command.prepositions['to'] = 'at/to';
+$.utils.command.prepositions['in front of'] = 'in front of';
+$.utils.command.prepositions['in'] = 'in/inside/into';
+$.utils.command.prepositions['inside'] = 'in/inside/into';
+$.utils.command.prepositions['into'] = 'in/inside/into';
+$.utils.command.prepositions['on top of'] = 'on top of/on/onto/upon';
+$.utils.command.prepositions['on'] = 'on top of/on/onto/upon';
+$.utils.command.prepositions['onto'] = 'on top of/on/onto/upon';
+$.utils.command.prepositions['upon'] = 'on top of/on/onto/upon';
+$.utils.command.prepositions['over'] = 'over';
+$.utils.command.prepositions['through'] = 'through';
+$.utils.command.prepositions['under'] = 'under/underneath/beneath';
+$.utils.command.prepositions['underneath'] = 'under/underneath/beneath';
+$.utils.command.prepositions['beneath'] = 'under/underneath/beneath';
+$.utils.command.prepositions['behind'] = 'behind';
+$.utils.command.prepositions['beside'] = 'beside';
+$.utils.command.prepositions['for'] = 'for/about';
+$.utils.command.prepositions['about'] = 'for/about';
+$.utils.command.prepositions['is'] = 'is';
+$.utils.command.prepositions['as'] = 'as';
+$.utils.command.prepositions['off'] = 'off/off of';
+$.utils.command.prepositions['off of'] = 'off/off of';
+
+$.utils.command.prepositionsRegExp = new RegExp(
+    '(?:^|\\s)(with|using|upon|underneath|under|to|through|over|out +of|' +
+    'onto|on +top +of|on|off +of|off|is|into|inside|in +front +of|' +
+    'in|from +inside|from|for|beside|beneath|behind|at|as|about)(?:\\s|$)');
+
+$.utils.command.parse = function(command, user) {
+  // Parse a user's command into components:
+  // verb, argstr, args, dobjstr, dobj, prepstr, iobjstr, iobj
+  var args = command.split(/ +/);
+  var verb = args.shift();
+  if (!verb) {
+    return undefined;
+  }
+  var argstr = command.substring(command.indexOf(verb) + verb.length + 1);
+  var dobjstr = '';
+  var dobj = null;
+  var iobjstr = '';
+  var iobj = null;
+  var prepstr = '';
+  var m = argstr.match($.utils.command.prepositionsRegExp);
+  if (m) {
+    prepstr = m[1];
+    var i = argstr.indexOf(prepstr);
+    if (i === -1) {
+      throw RangeError('Impossible preposition: ' + prepstr);
+    }
+    dobjstr = argstr.substring(0, i - 1);
+    iobjstr = argstr.substring(i + prepstr.length + 1);
+    prepstr = prepstr.replace(/ +/g, ' ');
```

Comment:
This would make it more difficult to support a feature Lambda Moo has which is quoted objects: `say "I'm at home" to Max`
In this case we want the prep to be "to" not "at".